chapter10/index.adoc

```diff
@@ -0,0 +1,484 @@
+
+=== Structure of a JWT
+
+A JWT consists of three Base64 encoded parts, separated by dots (+.+): 
+
+[source]
+----
+<Header>.<Payload>.<Signature>
+----
+
+- *Header* -  It contains metadata about the token, such as token type (type: “JWT”) and signing algorithm ( alg: “RS256” for RSA-SHA256). 
+
+[source, json]
+----
+{
+  "alg": "RS256",
+  "typ": "JWT"
+}
+----
+- *Payload* - It contains claims which are key-value pairs representing data about the user such as roles and expiration.
+Example of claims in a JWT payload:
+
+[source, json]
+----
+{
+  "iss": "https://io.microprofile.com/issuer",
+  "sub": "user1",
+  "exp": 1735689600,
+  "iat": 1735686000,
+  "aud": "my-audience",
+  "groups": ["user", "admin"]
+}
+----
+
+- *Signature* - A digital signature that is used to verify the token’s integrity by combining the encoded header, payload, and a secret/private key.
+
+Example JWT Token:
+[source]
+----
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ1c2VyMTIzIiwiaXNzIjoiaHR0cHM6Ly9hdXRoLmV4YW1wbGUuY29tIiwiZ3JvdXBzIjpbInVzZXIiLCJhZG1pbiJdfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+----
+
+==== Types of Claims in MicroProfile JWT
+
+Claims in JWTs can be categorized into two types:
+
+===== Standard Claims
+
+These are predefined claims with specific meanings, as defined by the JWT specification. MicroProfile JWT recognizes and validates these claims during token validation. Some commonly used standard claims include:
+
+|===
+|*Claim*|*Description*|*Example*
+
+|`iss`|Issuer, the entity that issued the JWT (e.g., an authentication server)|`"iss": "https://io.microprofile.com/issuer"`
+|`sub`|Subject, the principal (user or service) that the JWT is about.|`"sub": "user1"`
+|`aud`|Audience, the intended recipients of the token (e.g., specific microservices).|`"aud": "order-service"`
+|`exp`|Expiration time|`"exp": 1735689600`
+|`nbf`|not before|`"nbf": 1735686000`
+|`iat`|issued at, when that token was issued.|`"iat": 1735686000`
+|`jti`|Unique JWT token identifier|`"jti": "a1b2c3d4"`
+|`groups`|Groups, list of roles or users allowed to access the resource |`["user", "admin"]`
+|===
+
+===== Custom Claims
+
+These are application-specific claims that provide additional information about the user or entity. They can be used to extend authorization logic with application-specific claims (e.g., `department`, `region`). Custom claims are not part of the JWT specification but are often used to include domain-specific data, such as user preferences, tenant IDs, or other metadata. MicroProfile JWT allows developers to access these claims programmatically.
+
+=== Benefits of JWTs in Microservices?
+
+JWTs are a foundational technology for securing distributed systems due to their compact, self-contained, and stateless nature. These characteristics make them particularly well-suited for microservices architectures, where scalability, interoperability, and decentralized security are critical.
```

Comment:
```suggestion
```

This sounds too complex, example, services are stateless but JWTs can not be described as stateless. Neither they are compact. A simple sentence that you added below is great IMHO